sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1915,6 +1927,11 @@ class ReadFromBigQuery(PTransform):
         that dataset, and will remove it once it is not needed. Job needs access
         to create and delete tables within the given dataset. Dataset name
         should *not* start with the reserved prefix `beam_temp_dataset_`.
+    query_priority (BigQueryQueryPriority): By default, this transform runs
```

Comment:
I am wondering can we make a batch priority by default in ReadFromBigQuery? What would be the use case for the user to make it interactive? I just want to make sure that the parameter that got exposed are really necessary for the end user and it confirms with style guide

https://beam.apache.org/contribute/ptransform-style-guide/

Reply:
This PR does make the priority BATCH by default.

If you're asking about whether or not we should expose a `query_priority` parameter, I don't have strong opinions either way. My original commit (https://github.com/apache/beam/pull/15536/commits/b66e4b1361a6ccc0f5887777b0926630c7efc59e) makes it non-configurable, but after seeing that it's [configurable](https://beam.apache.org/releases/javadoc/2.32.0/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.TypedRead.html#withQueryPriority-org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.TypedRead.QueryPriority-) in the Java BigQueryIO, I decided to make the Python side consistent.

Reply:
never mind, my comment. I was thinking initially that the user doesn't need to be aware of this parameter. There's no point executing the query if there's not enough resources. Thus, batch priority for dataflow job makes better sense.

Reply:
thanks for checking all of these - I was not sure about adding the extra parameter, bu since Java exposes it, I think it makes sense for it to be exposed by Python as well.